### ISSUE TITLE
Cannot setup.py install on Ubuntu focal (20.04)

### ISSUE THREAD

**Issue by fbauzac** [pygraphviz/pygraphviz#251]
The current setup_extra.py code handles "dpkg -S" output that look like:

```
libgraphviz-dev: /usr/include/graphviz/geom.h
```

But it does not work with newer dpkg versions which have a slightly
different output:

```
libgraphviz-dev:amd64: /usr/include/graphviz/geom.h
```

The result is that I can't install it on Ubuntu focal (20.04):

```shell
$ setup.py install
running install
Trying dpkg
Trying pkg-config
Failed to find pkg-config
Trying dotneato-config
Failed to find dotneato-config
Failed to find dotneato-config

Your Graphviz installation could not be found.
```

**Comment by fbauzac** [comment]
This will be fixed with my pull request https://github.com/pygraphviz/pygraphviz/pull/250

**Comment by johnthagen** [comment]
I am seeing similar problems trying to install `pygraphviz` into an Ubuntu 20.04 Docker image:

```
    Running setup.py install for pygraphviz: started
    Running setup.py install for pygraphviz: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: .tox/py38/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-jbc93xrr/pygraphviz/setup.py'"'"'; __file__='"'"'/tmp/pip-install-jbc93xrr/pygraphviz/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-5agtnz77/install-record.txt --single-version-externally-managed --compile --install-headers .tox/py38/include/site/python3.8/pygraphviz
         cwd: /tmp/pip-install-jbc93xrr/pygraphviz/
    Complete output (31 lines):
    running install
    Trying dpkg
    Trying pkg-config
    Failed to find pkg-config
    Trying dotneato-config
    Failed to find dotneato-config
    Failed to find dotneato-config
    
    Your Graphviz installation could not be found.
    
            1) You don't have Graphviz installed:
               Install Graphviz (http://graphviz.org)
    
            2) Your Graphviz package might incomplete.
               Install the binary development subpackage (e.g. libgraphviz-dev or similar.)
    
            3) You are using Windows
               There are no PyGraphviz binary packages for Windows but you might be
               able to build it from this source.  See
               http://networkx.lanl.gov/pygraphviz/reference/faq.html
    
            If you think your installation is correct you will need to manually
            provide path to graphviz include and library. For example:
    
            pip install pygraphviz --install-option="--include-path=/usr/include/graphviz" --install-option="--library-path=/usr/lib/graphviz/"
    
            The current setting of library_dirs and include_dirs is:
    library_dirs=None
    include_dirs=None
    
    error: Error locating graphviz.
    ----------------------------------------
```

**Comment by dschult** [comment]
The solution depends on where your Graphviz installation put the libraries.
I think the help message here is pretty clear. Is there a question you have about how to proceed?

**Comment by johnthagen** [comment]
On Ubuntu 18.04 everything works out of the box, so I thought that was the expected situation for 20.04 and perhaps the `setup.py` needed to be updated to support 20.04.

I will try to recreate a minimal test case in Docker to help reproduce the issue.

**Comment by johnthagen** [comment]
### Ubuntu 18.04

This works out of the box:

```Dockerfile
FROM ubuntu:18.04

ENV DEBIAN_FRONTEND=noninteractive

RUN apt-get update && \
    apt-get install -y python3-pip python3-dev graphviz-dev

RUN /usr/bin/python3 -m pip install --upgrade pip && \
    pip install pygraphviz
```

#### Output

```
$ docker build .
...
Step 3/3 : RUN /usr/bin/python3 -m pip install --upgrade pip &&     pip install pygraphviz
 ---> Running in b4a358652f03
Collecting pip
  Downloading https://files.pythonhosted.org/packages/4e/5f/528232275f6509b1fff703c9280e58951a81abe24640905de621c9f81839/pip-20.2.3-py2.py3-none-any.whl (1.5MB)
Installing collected packages: pip
  Found existing installation: pip 9.0.1
    Not uninstalling pip at /usr/lib/python3/dist-packages, outside environment /usr
Successfully installed pip-20.2.3
Collecting pygraphviz
  Downloading pygraphviz-1.6.zip (117 kB)
Building wheels for collected packages: pygraphviz
  Building wheel for pygraphviz (setup.py): started
  Building wheel for pygraphviz (setup.py): finished with status 'done'
  Created wheel for pygraphviz: filename=pygraphviz-1.6-cp36-cp36m-linux_x86_64.whl size=155228 sha256=e5a513d80a46aaa9b00366ece49841dc4c26161ad4bf7d0a29f08ebcdfff8063
  Stored in directory: /root/.cache/pip/wheels/ae/dd/00/72ba3f6cec9d5c207e8dde77052acf6c27a58afefa0775ce04
Successfully built pygraphviz
Installing collected packages: pygraphviz
Successfully installed pygraphviz-1.6
Removing intermediate container b4a358652f03
 ---> fe0c02fa39af
Successfully built fe0c02fa39af
```

### Ubuntu 20.04

```Dockerfile
FROM ubuntu:20.04

ENV DEBIAN_FRONTEND=noninteractive

RUN apt-get update && \
    apt-get install -y python3-pip python3-dev graphviz-dev

RUN /usr/bin/python3 -m pip install --upgrade pip && \
    pip install pygraphviz
```


#### Output

```
$ docker build .
...
 ERROR: Failed building wheel for pygraphviz
  Running setup.py clean for pygraphviz
Failed to build pygraphviz
Installing collected packages: pygraphviz
    Running setup.py install for pygraphviz: started
    Running setup.py install for pygraphviz: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-m8b77fdo/pygraphviz/setup.py'"'"'; __file__='"'"'/tmp/pip-install-m8b77fdo/pygraphviz/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-9r10v_4r/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.8/pygraphviz
         cwd: /tmp/pip-install-m8b77fdo/pygraphviz/
    Complete output (31 lines):
    running install
    Trying dpkg
    Trying pkg-config
    Failed to find pkg-config
    Trying dotneato-config
    Failed to find dotneato-config
    Failed to find dotneato-config
    
    Your Graphviz installation could not be found.
    
            1) You don't have Graphviz installed:
               Install Graphviz (http://graphviz.org)
    
            2) Your Graphviz package might incomplete.
               Install the binary development subpackage (e.g. libgraphviz-dev or similar.)
    
            3) You are using Windows
               There are no PyGraphviz binary packages for Windows but you might be
               able to build it from this source.  See
               http://networkx.lanl.gov/pygraphviz/reference/faq.html
    
            If you think your installation is correct you will need to manually
            provide path to graphviz include and library. For example:
    
            pip install pygraphviz --install-option="--include-path=/usr/include/graphviz" --install-option="--library-path=/usr/lib/graphviz/"
    
            The current setting of library_dirs and include_dirs is:
    library_dirs=None
    include_dirs=None
    
    error: Error locating graphviz.
```

**Comment by johnthagen** [comment]
I was able to work around this by installing [`pkg-config`](https://stackoverflow.com/a/15667958/):

```Dockerfile
FROM ubuntu:20.04

ENV DEBIAN_FRONTEND=noninteractive

RUN apt-get update && \
    apt-get install -y python3-pip python3-dev pkg-config graphviz-dev

RUN /usr/bin/python3 -m pip install --upgrade pip && \
    pip install pygraphviz
```

**Comment by fbauzac** [comment]
The setup.py (via setup_extra.py) normally autodetects graphviz when installed on a debian-like operating system by calling "dpkg -S".  This worked well until the output format of this command changed slightly in e.g. Ubuntu focal (20.04).  Please check my pull request above which fixes the problem.

**Comment by rossbar** [comment]
Again, thanks for chasing this down @fbauzac - this should be fixed in pygraphviz 1.7, so I will close it. If there are still problems installing pygraphviz on Ubuntu or [the installation docs](https://pygraphviz.github.io/documentation/stable/install.html#ubuntu-and-debian) don't adequately cover you use-case, please open a new issue.
